site-src/geps/gep-724.md

```diff
@@ -0,0 +1,494 @@
+
+#### Removed
+
+```go
+  // Selector specifies a set of route labels used for selecting
+  // routes to associate with the Gateway. If this Selector is defined,
+  // only routes matching the Selector are associated with the Gateway.
+  // An empty Selector matches all routes.
+  //
+  // Support: Core
+  //
+  // +optional
+  Selector *metav1.LabelSelector `json:"selector,omitempty"`
+```
+
+### Routes
+
+On Routes, we remove the `RouteGateways` struct and replace it with a list of
+parent references to attach to.
+
+#### Removed
+From Route Specs:
+```go
+  // Gateways defines which Gateways can use this Route.
+  //
+  // +optional
+  // +kubebuilder:default={allow: "SameNamespace"}
+  Gateways *RouteGateways `json:"gateways,omitempty"`
+```
+
+And the structs that references:
+```go
+// RouteGateways defines which Gateways will be able to use a route. If this
+// field results in preventing the selection of a Route by a Gateway, an
+// "Admitted" condition with a status of false must be set for the Gateway on
+// that Route.
+type RouteGateways struct {
+  // Allow indicates which Gateways will be allowed to use this route.
+  // Possible values are:
+  // * All: Gateways in any namespace can use this route.
+  // * FromList: Only Gateways specified in GatewayRefs may use this route.
+  // * SameNamespace: Only Gateways in the same namespace may use this route.
+  //
+  // +optional
+  // +kubebuilder:validation:Enum=All;FromList;SameNamespace
+  // +kubebuilder:default=SameNamespace
+  Allow *GatewayAllowType `json:"allow,omitempty"`
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // AttachTo defines the parent resources this Route should be attached to. The
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  AttachTo []AttachRef `json:"gateways,omitempty"`
+```
+
+And the struct that references:
+```go
+// AttachRef identifies an API object to attach to. The only kind of target
+// resource with "Core" support is Gateway. This API may be extended in the
+// future to support additional kinds of parent resources such as Routes.
+type AttachRef struct {
+  // Group is the group of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=networking.x-k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
```

Comment:
Yeah completely agree here, I came very close to adding a Gateway default as well but ultimately decided against it.

Reply:
is it backwards compatible to add a default later? Seems like it should be?

I don't have a strong preference, but suspect 99% of use cases will be `Gateway` so I could see folks liking the defaulting. Good to be able to add it later if its an issue

Reply:
> is it backwards compatible to add a default later?

I don't think so but I'm not sure.

I'd recommend to adding a default here, I expect a large majority would be gateway.

Reply:
I lean towards explicitness on this one as well. +1 to no default.

Reply:
If I'm interpreting this correctly, the current votes for setting Gateway as default here are:

@youngnick: No
@Miciah: Leaning yes
@hbagdi: Yes
@howardjohn: Fine either way
@jpeach: Leaning no

If I've interpreted that correctly, it seems like we're effectively tied. Fortunately we can change defaults post-v1alpha2 release. Although I initially omitted a default here, I'm leaning towards adding one here since it does represent 99% of use cases. If anyone feels strongly that we should leave out a default, I'm happy to leave it off though.

Reply:
I can't figure out what this thread refers to any more :joy:

Reply:
Haha this is in the context of `route.spec.parentRefs.kind` and the question is if we should default it to "Gateway". It seems like no one feels overly strongly on this one, but if anyone does I'm happy to go that way. At present, I'm leaning towards leaving it with a "Gateway" default because that represents such a large majority of use cases.